habitat_baselines/common/base_trainer.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+
+from habitat_baselines.common.trainer_registry import trainer_registry
+
+
+class BaseTrainer(ABC):
```

Comment:
We should think about more useful abstract methods to add to the `BaseTrainer`, @JasonJiazhiZhang if possible can you communicate on the PR what are the implementations you referred to when building out the base trainer and what you eventually ended up using.

Reply:
Sure, this draft PR is precisely for discussions like this. This BaseTrainer structure is inspired mainly by the base_class.py in [stable-baselines](https://github.com/hill-a/stable-baselines/blob/master/stable_baselines/common/base_class.py). I will put all the sources I looked in the description.